### ISSUE TITLE
stack rm fails to remove containers that are in restart loop

### ISSUE THREAD

**Issue by matti** [kontena/kontena#2608]
stack with two services, the other service fails to start --> kontena stack rm --> docker keeps restarting the service for ~ 2 minutes until agent (?) removes the container

**Comment by jakolehm** [comment]
Delay is most likely caused by `wait_for_port` option. Agent waits max 5 mins port to open.. after that it will read desired state again and remove service instance (container).

**Comment by matti** [comment]
delay is veried to be `wait_for_port`

**Comment by SpComb** [comment]
Fixed in 1.4 by #2776, the `kontena stack rm` now has ~immediate effect even if the initial deploy is in the `wait_for_port`.

```yaml
stack: test/waitforport
services:
  redis:
    image: redis
    deploy:
      wait_for_port: 6378
```

```
$ bundle exec bin/kontena stack install test/waitforport.yaml 
 [done] Creating stack waitforport      
 [done] Triggering deployment of stack waitforport     
 [done] Waiting for deployment to start     
 [fail] Deploying service redis     ^C
$ bundle exec bin/kontena stack rm --force waitforport
 [done] Removing stack waitforport      
```

```
D, [2018-01-09T10:04:59.011381 #1] DEBUG -- Kontena::Workers::ServicePodWorker: state of waitforport.redis-1: running
D, [2018-01-09T10:04:59.087148 #1] DEBUG -- Kontena::Workers::ServicePodWorker: state is in-sync: running
I, [2018-01-09T10:04:59.104334 #1]  INFO -- Kontena::Workers::ServicePodWorker: waiting until container /waitforport.redis-1 port 10.81.128.33:6378 is responding
D, [2018-01-09T10:05:02.121406 #1] DEBUG -- Kontena::Workers::ServicePodWorker: waiting 4.0s of 300.0s until: container /waitforport.redis-1 port 10.81.128.33:6378 is responding
D, [2018-01-09T10:05:03.124945 #1] DEBUG -- Kontena::Workers::ServicePodWorker: waiting 5.0s of 300.0s until: container /waitforport.redis-1 port 10.81.128.33:6378 is responding
D, [2018-01-09T10:05:03.749324 #1] DEBUG -- Kontena::Workers::ServicePodWorker: waiting 9.8s of 300.0s until: container /waitforport.redis-1 port 10.81.128.33:6378 is responding
D, [2018-01-09T10:05:07.676836 #1] DEBUG -- Kontena::Workers::ServicePodWorker: waiting 9.6s of 300.0s until: container /waitforport.redis-1 port 10.81.128.33:6378 is responding
D, [2018-01-09T10:05:11.312097 #1] DEBUG -- Kontena::Workers::ServicePodWorker: waiting 17.4s of 300.0s until: container /waitforport.redis-1 port 10.81.128.33:6378 is responding

D, [2018-01-09T10:05:13.135616 #1] DEBUG -- Kontena::RpcServer: rpc notification: Kontena::Rpc::ServicePodsApi#notify_update ["stop"]
D, [2018-01-09T10:05:13.146517 #1] DEBUG -- Kontena::RpcServer: rpc notification: Kontena::Rpc::ServicePodsApi#notify_update ["terminate"]

D, [2018-01-09T10:05:13.187542 #1] DEBUG -- Kontena::Workers::ServicePodWorker: state of waitforport.redis-1: terminated
I, [2018-01-09T10:05:13.224462 #1]  INFO -- Kontena::Workers::ServicePodWorker: terminating waitforport.redis-1
I, [2018-01-09T10:05:13.241620 #1]  INFO -- Kontena::ServicePods::Terminator: terminating service: /waitforport/redis-1

W, [2018-01-09T10:05:13.666536 #1]  WARN -- Kontena::Workers::ServicePodWorker: failed to sync waitforport.redis-1 at 2018-01-09 10:04:50 UTC: service stopped
W, [2018-01-09T10:05:13.668672 #1]  WARN -- Kontena::Workers::ServicePodWorker: service stopped (RuntimeError)
/app/lib/kontena/workers/service_pod_worker.rb:164:in `check_starting!'
/app/lib/kontena/workers/service_pod_worker.rb:184:in `block in wait_for_port'
/app/lib/kontena/helpers/wait_helper.rb:33:in `block in wait_until'
/app/lib/kontena/helpers/wait_helper.rb:32:in `loop'
/app/lib/kontena/helpers/wait_helper.rb:32:in `wait_until'
/app/lib/kontena/helpers/wait_helper.rb:63:in `wait_until!'
/app/lib/kontena/workers/service_pod_worker.rb:183:in `wait_for_port'
/app/lib/kontena/workers/service_pod_worker.rb:140:in `apply'
/usr/lib/ruby/gems/2.4.0/gems/celluloid-0.17.3/lib/celluloid/calls.rb:28:in `public_send'
/usr/lib/ruby/gems/2.4.0/gems/celluloid-0.17.3/lib/celluloid/calls.rb:28:in `dispatch'
/usr/lib/ruby/gems/2.4.0/gems/celluloid-0.17.3/lib/celluloid/call/async.rb:7:in `dispatch'
/usr/lib/ruby/gems/2.4.0/gems/celluloid-0.17.3/lib/celluloid/cell.rb:50:in `block in dispatch'
/usr/lib/ruby/gems/2.4.0/gems/celluloid-0.17.3/lib/celluloid/cell.rb:76:in `block in task'
/usr/lib/ruby/gems/2.4.0/gems/celluloid-0.17.3/lib/celluloid/actor.rb:339:in `block in task'
/usr/lib/ruby/gems/2.4.0/gems/celluloid-0.17.3/lib/celluloid/task.rb:44:in `block in initialize'
/usr/lib/ruby/gems/2.4.0/gems/celluloid-0.17.3/lib/celluloid/task/fibered.rb:14:in `block in create'
D, [2018-01-09T10:05:13.740469 #1] DEBUG -- Kontena::Workers::ServicePodWorker: sync waitforport/redis-1 state at 2018-01-09 10:04:50 UTC: {:service_id=>"5a5493c19bb44d00089f2b49", :instance_number=>1, :rev=>"2018-01-09 10:04:50 UTC", :state=>"stopped", :error=>"RuntimeError: service stopped"}
W, [2018-01-09T10:05:13.744501 #1]  WARN -- Kontena::Workers::ServicePodWorker: failed to sync waitforport.redis-1 at 2018-01-09 10:04:50 UTC: service stopped
```
